internal/cloudsql/instance.go

```diff
@@ -225,6 +234,16 @@ func (c ConnectionInfo) Addr(ipType string) (string, error) {
 // TLSConfig constructs a TLS configuration for the given connection info.
 func (c ConnectionInfo) TLSConfig() *tls.Config {
 	pool := x509.NewCertPool()
+	if c.ServerCaMode == "GOOGLE_MANAGED_CAS_CA" {
+		pool.AppendCertsFromPEM(c.ServerCaCertPem)
+		return &tls.Config{
+			ServerName:         c.DNSName,
+			Certificates:       []tls.Certificate{c.ClientCertificate},
+			RootCAs:            pool,
+			InsecureSkipVerify: false,
+			MinVersion:         tls.VersionTLS13,
+		}
+	}
 	pool.AddCert(c.ServerCaCert)
 	return &tls.Config{
```

Comment:
You will need to modify the `verifyPeerCertificateFunc()`. On like 287, it checks the certificate CommonName field against the expected instance name. This check should not occur when a custom SAN is in use.

Reply:
I believe for CAS instances we can rely on the default verification logic since the `InsecureSkipVerify` is false and we set the `ServerName` to the DNS name. We don't have to call `verifyPeerCertificateFunc()`.